forem/forem
# Remove deprecated package "postcss-next"

## What type of PR is this? (check all applicable)


## Description

This was deprecated a few years ago, and appears to be causing issues
with a newer (postcss v8) api.

https://moox.io/blog/deprecating-cssnext
https://github.com/csstools/postcss-preset-env/issues/32

This appears to fix https://github.com/forem/forem/issues/14052 but
was tested only by observing that the postcss command no
longer fails with an error, and not that browser ready css is
generated from the input or working for users.

~It's likely that the next step is to add back
postcss-preset-env (which does not include postcss-calc, which was I
think related to the broken parses and calling the older version of
postcss internally).~ nothing to do here - postcss-preset-env was included by webpacker automatically and is already configured.

Actually, postcss-preset-env was installed, and looked like it was configured, but was not added to a .postcssrc.yml file which was overriding the "default + loaded config" behavior and causing this to be skipped after I removed the postcss-cssnext plugin from the rc file. That's been changed in this PR (the config options from the postcss.config.js file were not brought into the yaml rc file) and once that was enabled I could confirm the postcss processor was running again (files were being parsed, and failing). 

A change to the max-width property of the runtime-banner element (used to suggest the native app to ios users) appears to have been the cause of the parse error, and an equivalent formulation has been used instead. I still don't understand what, between sass and postcss processing, was happening between calc(min(x,y)) to cause the error, but we don't do that anymore.

The error redirection is left disabled, I currently see two warnings (which are generated in a single file context but don't give the file being processed, it's not very usable but we can let it grow until it's a pain point again). 

Sure wish it told me what the file with 70K lines with the "start" value was.

```
 yarn run v1.22.5
 $ postcss public/assets/*.css -d public/assets

70881:3 ⚠  start value has mixed support, consider using flex-start instead [postcss-preset-env]                                                                      
71131:3 ⚠  start value has mixed support, consider using flex-start instead [postcss-preset-env]                                                                      

 Done in 4.47s.
```


## Related Tickets & Documents

fixes #14052

## QA Instructions, Screenshots, Recordings

The postcss script is called as a post precompile assets hook from rake, or separately executable from yarn.
The simplest proof of concept would be to run `yarn postcss`.

I don't know how to check that features supported or intended to be supported by the processor (which was erroring every time I looked at it) are actually being processed. I assume that's something like validating css loads correctly in some target set of browsers.

## Questions for reviewers

Do we want to remove the .postcssrc.yml file (and let the postcss.config.js file take effect)? 
Do we instead want to copy any of the config for postcss-preset-env into the postcssrc.yml file (options hash for plugin)?
Does the width/max-width correctly capture Fernando's intent for the `.runtime-banner`?
Are we comfortable letting the warnings show up again in the builds (the original issue leading to hiding them appears to have been a wall of warnings causing build failures due to overflow or timeouts).


### UI accessibility concerns?

Very possibly - if our accessibility was based on the css - however from what I can tell we're just removing broken software rather than changing some configuration that had been working correctly. Since it was just failing (from what I can see) I don't think we're any worse off.

## Added/updated tests?

- [x] No, and this is why: build config change

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

Probably spot check with browsershots or some other cross-platform test tool to make sure forem isn't failing to render correctly. 

## [optional] What gif best describes this PR or how it makes you feel?

![delete key](https://user-images.githubusercontent.com/1237369/123139473-dce2f900-d41b-11eb-8f09-4d00fcfdf094.png)
